### ISSUE TITLE
Save scientific work citations to a BibTeX file for ease of use in Doxygen

### ISSUE THREAD

**Issue by jhlegarreta** [InsightSoftwareConsortium/ITK#3662]
### Description

Citations to scientific works should be stored in a BibTeX file and Doxygen's `\cite` should be used to cite them in the documentation. Currently, there is a variety of syntaxes that are used to cite works across the documentation, ranging from no syntax highlight at all, code-style indentation, or paragraph, e.g.:
https://itk.org/Doxygen/html/classitk_1_1STAPLEImageFilter.html
https://itk.org/Doxygen/html/classitk_1_1ScalarChanAndVeseSparseLevelSetImageFilter.html
https://itk.org/Doxygen/html/classitk_1_1BSplineScatteredDataPointSetToImageFilter.html

### Expected information

Citatons should be hosted in a single BibTeX file so that they can easily be cross-referenced and reused across the documentation.

### Actual information

Citation styles are inconsistent and the same work might be cited differently/citations cannot be re-used.

### Versions

`master`

### Additional Information

Doxygen `\cite` command documentation:
https://doxygen.nl/manual/commands.html#cmdcite

The IJ works' bibtex files can actually be retrieved from the IJ website, so that would save some work. Maybe an script that gathers all references could be developed to have them all in a single file. The file could be dynamically updated e.g. every month.

IJ repository: https://github.com/InsightSoftwareConsortium/InsightJournal

**Comment by zivy** [comment]
A limitation of the html version generated bibliography is that the style is fixed and even when the doi is available as part of the bib file it is not included in the output. This functionality is a [longstanding feature request](https://github.com/doxygen/doxygen/issues/5450) in doxygen.

**Comment by jhlegarreta** [comment]
I see. Anyways, I believe consistency is important, so even if the DOI is not displayed by Doxygen, having all references in a BibTeX file would still be an improvement: easier to maintain, consistent style when being displayed, etc. Maybe the Doxygen folks can afford addressing the DOI issue at some point in the future.

**Comment by zivy** [comment]
Agreed. Using the bibtex format forces us to provide more complete and consistent citation information. Some of the existing textual references were partial and ambiguous, required some searching to pinpoint the referenced paper.

**Comment by albert-github** [comment]
One thing I saw is that when the "doi" is in the url there is a link to the "doi" see e.g. https://doc.cgal.org/latest/Arrangement_on_surface_2/citelist.html (ind the link is forwarded again to springer) where the bib file entry is:
```
@article{cgal:bfhks-apsca-10,
  author      = {Eric Berberich and Efi Fogel and Dan Halperin and
                 Michael Kerber and Ophir Setter},
  title       = {Arrangements on Parametric Surfaces {II}: Concretizations
                 and Applications},
  journal     = {Mathematics in Computer Science},
  publisher   = {Birkh\"{a}user Basel},
  issn        = {1661-8270},
  keyword     = {Mathematics},
  pages       = {67--91},
  volume      = {4},
  issue       = {1},
  url         = {https://dx.doi.org/10.1007/s11786-010-0043-4},
  year        = {2010}
}

```


Note: just a mention or a thumbs up does not trigger an email (so not known there is extra need for it), a friendly ping might help.

**Comment by zivy** [comment]
Thanks @albert-github for the pointer. Will duplicate the doi info in the url field so that the current doxygen config will result in hyperlinks.

**Comment by albert-github** [comment]
Isn't this fixed now by means of #5149

**Comment by dzenanz** [comment]
That PR states it is the first of multiple PRs to accomplish that. Some citations were converted, but not all.

**Comment by zivy** [comment]
@jhlegarreta, @dzenanz 
I just created the last PR for this issue. Note that this is a best-effort to identify all of the references throughout the code base. I'm sure there are references I missed and I intentionally did not include the Insight-Journal papers that are referenced throughout (their DOIs are listed so readily accessible to the reader).

**Comment by dzenanz** [comment]
Thank you Ziv!

**Comment by jhlegarreta** [comment]
Thanks for the effort @zivy.

>  I intentionally did not include the Insight-Journal papers that are referenced throughout (their DOIs are listed so readily accessible to the reader).

I would keep the issue open until we have the bandwidth to address those so that we never look back again to non-BibTeX refs in ITK.

**Comment by zivy** [comment]
@jhlegarreta, sure. I'll need to amend my last commit message so that merging the PR doesn't automatically close this issue.

**Comment by jhlegarreta** [comment]
> I'll need to amend my last commit message so that merging the PR doesn't automatically close this issue.

:+1: Maybe remove the 4/4 and final parts as well.

**Comment by zivy** [comment]
> 👍 Maybe remove the 4/4 and final parts as well.
Let me keep those. It's the final PR for me on this for a while (I didn't know what I was getting myself into when I assigned this issue to myself 😆 ).

**Comment by albert-github** [comment]
@zivy 

Nice work.
There are still some problems (besides the references to the Insight Journal Paper): 
- issue #5218 with proposed patch
- issue #5203 waiting for fix on ITK site or guidance what could be done (as I suspect that the problems are cased by clang-format)
- issue #5219 with proposed patch

**Comment by zivy** [comment]
@albert-github Thanks for identifying the issues and solutions. Will address these shortly.

**Comment by albert-github** [comment]
I've done a "small" grep on the sources to identify some more doi entries and created this bib file: [doi2bib.bib.tar.gz](https://github.com/user-attachments/files/18725645/doi2bib.bib.tar.gz)

Also thanks to https://github.com/schneiderfelipe/doi (especially the file: https://github.com/schneiderfelipe/doi/blob/master/src/doi2bib)


There were a few references for which I was unable to create a bibtex entry:
- https://doi.org/10.5281/zenodo.8136801
  Looks like to be a poster at SciPy 2023 conference, I copuldn't find a bibtex entry

- https://doi.org/10.54294/j8lsa66
  description in code doesn't help me anywhere further


- https://doi.org/10.54294/lf8u753
  description in code doesn't help me anywhere further

- https://doi.org/10.54294/olkmog2
  Looks like to be, is this correct?
  ```
  @article{McCormick2010
  author = "M. McCormick",
  title = "An Open Source, Fast Ultrasound B-Mode Implementation for Commodity Hardware",
  howpublished = "\url{http://hdl.handle.net/10380/3159}",
  year = 2010,
  month = 05,
  abstract = "This document describes an open source, high performance ultrasound B-Mode implementation based on the Insight Toolkit (ITK). ITK extensions are presented to calculate the radio-frequency (RF) signal envelope. A variety of 1D Fast Fourier Transform options are introduced including VNL, FFTW, and an OpenCL solution. Scan conversion is implemented for phased array or curvilinear transducers. The entire image processing pipeline is streamable to limit memory consumption during multi-frame or 3D acquisitions with the introduction of an itk::StreamingResampleImageFilter.
  ",
  institution = "University of Wisconsin-Madison",
  publisher = "The Insight Journal",
  doi = "10.54294/yjowe4",
  }
  ```

- https://doi.org/10.54294/olkmog9
  Looks like to be, is this correct?
  ```
  @article{Lowekamp2010,
    title = {A Streaming IO Base Class and Support for Streaming the MRC and VTK File Format},
    ISSN = {2327-770X},
    url = {http://dx.doi.org/10.54294/ufs19w},
    DOI = {10.54294/ufs19w},
    journal = {The Insight Journal},
    publisher = {NumFOCUS - Insight Software Consortium (ITK)},
    author = {Lowekamp,  Bradley and Chen,  David},
    year = {2010},
    month = jun
  }
  ```

**Comment by albert-github** [comment]
Independent of the above, I found 2 small problems, and thus incorrect link / no link, in the bib file, proposed patch: [diff.patch](https://github.com/user-attachments/files/18732759/diff.patch)

Citations with titles:
- `Image characterizations based on joint gray level—run length distributions`
- `Digital image thresholding, based on topological stable-state`

**Comment by blowekamp** [comment]
@albert-github I think McCormick2010 should reference the preferred DOI: https://doi.org/10.54294/yjowe4

Lowekamp2010 - looks right to me.

**Comment by dzenanz** [comment]
I turned this into a PR: #5234.

**Comment by albert-github** [comment]
Proposed patch for the 2 "olkmog" type of references in the code: [diff.patch](https://github.com/user-attachments/files/18734997/diff.patch).

**Comment by dzenanz** [comment]
> Independent of the above, I found 2 small problems, and thus incorrect link / no link, in the bib file, proposed patch: [diff.patch](https://github.com/user-attachments/files/18732759/diff.patch)
> 
> Citations with titles:
> 
> * `Image characterizations based on joint gray level—run length distributions`
> * `Digital image thresholding, based on topological stable-state`

Done via PR #5233.

**Comment by albert-github** [comment]
@dzenanz 
Yes I can imagine it takes some time  for you, would make my live a bit easier as well probably.
I'll think about it in the future.

**Comment by dzenanz** [comment]
Should this issue be closed now?

**Comment by zivy** [comment]
I think it should. Looks like all the papers including insight journal ones are listed, so the Doxygen is now consistently using the bibtex file.

**Comment by albert-github** [comment]
The bibtex list contains all entries mentioned in the documentation, but the entries ares still full in the documentation and there are no references to the bibliography list (and thus doxygen does not mention them in the reference list either).

So I don't think this issue can be closed

**Comment by blowekamp** [comment]
> FYI
> 
> In the doxygen master version 1.14.0 (d8e7438cbb37ef1dffa7ae69e22e42a1100466b9) am extension to the `\cite` command has been integrated and this will become generally available when doxygen 1.14.0 be released (e.g. instead of a number the author can be used as link).

The Dockerfile in the ITKDoxygen repo could be update to pull the recommended hash:
https://github.com/InsightSoftwareConsortium/ITKDoxygen/blob/main/Dockerfile#L20
